### PR TITLE
Add Cutout to geometry - fix VMobject().force_direction()

### PR DESCRIPTION
**Pull request by XorUnison**
## List of Changes
Added a class that allows for making shapes with an arbitrary number of cutouts. See the example in the documentation.
Also fixed a really sneaky bug in VMobject().force_direction().

## Motivation
The code is trivial yet the resulting class is very useful for a whole number of applications.

## Other comments
~Also once the docs are built properly~ Docs are looking good. I'd like some feedback on naming the class. I'm not sure `Cutout` is the best choice. Also I'll have to add this and my previous PRs to the changenotes.

### CODE REVIEW

File: manim/mobject/geometry.py
Comment (leotrs) [inline manim/mobject/geometry.py:1432]:
```suggestion
    def __init__(self, main_shape, *mobjects, **kwargs):
        VMobject.__init__(self, **kwargs)
        self.append_points(main_shape.get_points())
        if main_shape.get_direction() == "CW":
            sub_direction = "CCW"
        else:
            sub_direction = "CW"
        for mobject in mobjects:
            self.append_points(mobject.force_direction(sub_direction).get_points())
```

**Review by leotrs — CHANGES_REQUESTED** [review]
Amazing.

**Review by leotrs — APPROVED** [review]

**Comment by behackl** [comment]
This looks very interesting, thanks for including your code! Regarding naming: when looking at your example, path operations from usual vector graphics editors came to my mind; there this operation would probably rather be described as a "Difference" -- but I think in this context I actually prefer "Cutout".

I just have one question before I approve this: what happens if the cutout shape is not fully contained within the `main_shape`?

**Comment by XorUnison** [comment]
> This looks very interesting, thanks for including your code! Regarding naming: when looking at your example, path operations from usual vector graphics editors came to my mind; there this operation would probably rather be described as a "Difference" -- but I think in this context I actually prefer "Cutout".

Yeah, pretty much what I thought.
> I just have one question before I approve this: what happens if the cutout shape is not fully contained within the `main_shape`?

grafik
And that's perfectly as expected too. See `Cutout` here makes use of the direction methods I implemented earlier, whereas explicit subsets of the main shape basically get subtracted iff their direction is inverse. There is a solution to this, technically, but it'd be 1 or 2 orders of magnitude more complicated. Hence this really just works as a cutout, not a set theory difference.

**Comment by behackl** [comment]
> > I just have one question before I approve this: what happens if the cutout shape is not fully contained within the `main_shape`?
> 
> And that's perfectly as expected too. See `Cutout` here makes use of the direction methods I implemented earlier, whereas explicit subsets of the main shape basically get subtracted iff their direction is inverse. There is a solution to this, technically, but it'd be 1 or 2 orders of magnitude more complicated. Hence this really just works as a cutout, not a set theory difference.

Perfect; just wanted to make sure that this behaves as I would have expected after reading your implementation. I'll add one suggestion for a corresponding warning/hint in the documentation, otherwise this is good to go for me.

**Comment by XorUnison** [comment]
Well... if we're being really strict, this class sort of works like the set symmetric difference, but I'm sure it'll break down even from that under the right circumstances. So better to stick with `Cutout` over that too.

File: manim/mobject/geometry.py
Comment (behackl) [inline manim/mobject/geometry.py:1399]: It would be good to explicitly mention the case where `mobjects` are not fully included in `main_shape`.

```suggestion
    .. warning::

        Technically, this class behaves similar to a symmetric difference: if
        parts of the ``mobjects`` are not located within the ``main_shape``,
        these parts will be added to the resulting :class:`~.VMobject`.

    Parameters
```

File: manim/mobject/geometry.py
Comment (behackl) [inline manim/mobject/geometry.py:1404]:
```suggestion
        The smaller shapes which are to be cut out of the ``main_shape``.
```

**Review by behackl — APPROVED** [review]
Two minor suggestions, as discussed, please consider them before merging. Rest LGTM. :-)

**Pull request merged**
